src/util/platform.h

```diff
@@ -40,4 +40,17 @@
   } while (0)
 #endif
 
+// Translate the MSVC _M_AMD64, _M_X64 and _M_IX86_FP to gcc style defines for SSE and SSE2 instruction sets
+#if defined(_M_AMD64) || defined(_M_X64)
+#define __SSE__
+#define __SSE2__
+#elif defined(_M_IX86_FP)
+#if _M_IX86_FP >= 1
+#define __SSE__
+#endif
+#if _M_IX86_FP == 2
+#define __SSE2__
+#endif
+#endif
+
```

Comment:
Doing this at the preprocessor Level inside a header file that you may forget to include is hacky IMHO.

According to [the documentation](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros#microsoft-specific-predefined-macros):
> **_M_IX86_FP**
>
> Expands to a value indicating which `/arch` compiler option was used:
> - 0 if `/arch:IA32` was used. 
> - 1 if `/arch:SSE` was used.
> -2 if `/arch:SSE2` was used. This value is the default if `/arch` was not specified.

So I think we should be able to set the defines from the `CMakeLists.txt`, as before.

Reply:
Moved to CMakeList.txt now.